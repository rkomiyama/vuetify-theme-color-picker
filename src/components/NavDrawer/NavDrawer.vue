<template>
  <v-navigation-drawer
    :value="$attrs.value"
    @input="$emit('input:drawer', $event)"
    mobile-break-point="960"
    app
  >
    <v-layout class="d-block">
      <NavDrawerItems />
    </v-layout>
  </v-navigation-drawer>
</template>

<script>
import NavDrawerItems from "./Items/NavDrawerItems";

export default {
  name: "NavDrawer",
  components: {
    NavDrawerItems
  }
};
</script>
