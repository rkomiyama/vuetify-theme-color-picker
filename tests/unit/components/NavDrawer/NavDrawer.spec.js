import Vue from 'vue';
import Vuetify from 'vuetify';
import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer';
import NavDrawer from '@/components/NavDrawer/NavDrawer.vue'

Vue.use(Vuetify)
let wrapper

describe('NavDrawer.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(NavDrawer)
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('is a Vue instance', () => {
    const wrapper = shallowMount(NavDrawer)
    expect(wrapper.isVueInstance()).toBeTruthy();
  })

  it('NavDrawer renders the same HTML', () => {
    const renderer = createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
    if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
})
