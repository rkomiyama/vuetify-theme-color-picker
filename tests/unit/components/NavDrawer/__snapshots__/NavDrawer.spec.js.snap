// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDrawer.vue NavDrawer renders the same HTML 1`] = `
<v-navigation-drawer-stub app="true" height="100vh" miniVariantWidth="80" mobileBreakPoint="960" src="" tag="nav" width="256">
  <v-layout-stub tag="div" class="d-block">
    <NavDrawerTitle-stub></NavDrawerTitle-stub>
    <NavDrawerItems-stub></NavDrawerItems-stub>
  </v-layout-stub>
</v-navigation-drawer-stub>
`;
